src/baseApi.ts

```diff
@@ -4,16 +4,31 @@ import { ApiResponse } from './utils/response/apiResponse';
 const DEFAULT_API_VERSION = 3;
 const DEFAULT_HOST = 'go.timetac.com';
 
+export type Tokens = {
+  accessToken: string;
+  refreshToken: string;
+};
+
+type onTokenRefreshedCallback = (tokens: Tokens) => void;
+
+export type ApiState = {
+  refreshingToken: false | Promise<AxiosResponse<any>>;
+};
+
 export type ApiConfig = {
   host?: string;
   version?: number;
   account?: string;
   accessToken?: string;
   refreshToken?: string;
+  clientId?: string;
+  clientSecret?: string;
+  onTokenRefreshedCallback?: onTokenRefreshedCallback;
+  autoRefreshToken?: boolean;
 };
 
 export default abstract class BaseApi {
-  public readonly resourceName: String = 'placeholder';
+  public readonly resourceName: String = '';
```

Comment:
I believe we can make this abstract, then we get an error in case we forget to implement that in the subclass
```suggestion
  public readonly abstract resourceName: String;
```